core/src/main/java/com/scalar/db/storage/jdbc/JdbcDatabaseAdmin.java

```diff
@@ -76,4 +189,172 @@ public void close() {
       LOGGER.error("failed to close the dataSource", e);
     }
   }
+
+  private void createSchemaIfNotExists(Connection connection, String schema) throws SQLException {
+    switch (rdbEngine) {
+      case MYSQL:
+      case POSTGRESQL:
+        execute(
+            connection,
+            "CREATE SCHEMA IF NOT EXISTS " + enclose(getFullNamespaceName(schemaPrefix, schema)));
+        break;
+      case SQL_SERVER:
+        try {
+          execute(
+              connection, "CREATE SCHEMA " + enclose(getFullNamespaceName(schemaPrefix, schema)));
+        } catch (SQLException e) {
+          // Suppress the exception thrown when the schema already exists
+          if (e.getErrorCode() != 2714) {
+            throw e;
+          }
+        }
+        break;
+      case ORACLE:
+        try {
+          execute(
+              connection,
+              "CREATE USER "
+                  + enclose(getFullNamespaceName(schemaPrefix, schema))
+                  + " IDENTIFIED BY \"oracle\"");
+        } catch (SQLException e) {
+          // Suppress the exception thrown when the user already exists
+          if (e.getErrorCode() != 1920) {
+            throw e;
+          }
+        }
+        execute(
+            connection,
+            "ALTER USER "
+                + enclose(getFullNamespaceName(schemaPrefix, schema))
+                + " quota unlimited on USERS");
+        break;
+    }
+  }
+
+  @SuppressFBWarnings("SQL_NONCONSTANT_STRING_PASSED_TO_EXECUTE")
+  private void createTableInternal(
+      Connection connection, String schema, String table, TableMetadata metadata)
+      throws SQLException {
+    String createTableStatement =
+        "CREATE TABLE "
+            + enclosedFullTableName(getFullNamespaceName(schemaPrefix, schema), table, rdbEngine)
+            + "(";
+    LinkedHashSet<String> sortedColumnNames =
+        Sets.newLinkedHashSet(
+            Iterables.concat(
+                metadata.getPartitionKeyNames(),
+                metadata.getClusteringKeyNames(),
+                metadata.getColumnNames()));
```

Comment:
Sorry, I missed it in the previous review. `metadata.getColumnNames()` should contain the primary key columns (the partition key and the clustering key columns). So we don't need to concatenate them here.

Reply:
Right, I did this to order the columns creation by `partition key >> clustering key >>  other columns` as I thought it was some sort of convention. It's better not to order it ?

Reply:
Ah okay, understood. So let's keep it. Thanks.